cgrok/grok_pattern.c

```diff
@@ -76,10 +76,14 @@ int grok_patterns_import_from_file(const grok_t *grok, const char *filename) {
   memset(buffer, 0, filesize);
   bytes = fread(buffer, 1, filesize, patfile);
   if (bytes != filesize) {
-    grok_log(grok, LOG_PATTERNS, "Unable to open '%s' for reading: %s",
-             filename, strerror(errno));
-    fprintf(stderr, "Expected %zd bytes, but read %zd.", filesize, bytes);
-    return GROK_ERROR_UNEXPECTED_READ_SIZE;
+    // Using ftell on a text file, we may get less bytes than we expect.
```

Comment:
On Windows in particular, opening a file with "r" and using `ftell` to get the length isn't 100% accurate - there may be fewer characters, but never more. Since it gives an upper bound I rewrote this to check for EOF, and error only if we didn't read the whole file.